File: 3ra Practica de 121/GENERICIDAD/1-caja/codigos/Caja.py
class Caja:
    def __init__(self):
        self.contenido = None

    def guardar(self, nuevo_contenido):
        self.contenido = nuevo_contenido

    def obtener(self):
        return self.contenido

if __name__ == "__main__":
    caja_str = Caja()
    caja_str.guardar("Hola Mundo")

    caja_int = Caja()
    caja_int.guardar(42)

    print("Caja String:", caja_str.obtener())
    print("Caja Int:", caja_int.obtener())

File: 3ra Practica de 121/GENERICIDAD/1-caja/codigos/Caja.java
public class Caja<T> {
    private T contenido;

    public void guardar(T nuevoContenido) {
        this.contenido = nuevoContenido;
    }

    public T obtener() {
        return this.contenido;
    }

    public static void main(String[] args) {
        Caja<String> cajaString = new Caja<>();
        cajaString.guardar("Hola Mundo");

        Caja<Integer> cajaInt = new Caja<>();
        cajaInt.guardar(42);

        System.out.println("Caja String: " + cajaString.obtener());
        System.out.println("Caja Int: " + cajaInt.obtener());
    }
}